R/git.R

```diff
@@ -71,3 +71,18 @@ git_pull <- function(root = NULL) {
   orderly_log("git", "pull")
   git_run("pull", root = root, check = TRUE)
 }
+
+git_ref_is_valid <- function(ref, root) {
+  if (is.null(ref)) {
+    return(invisible(TRUE))
+  }
+  tryCatch(
+    ## ref can be either a branch name or a git hash, git_commit_id will
+    ## give us full ID if ref is valid, if not throws an error.
+    gert::git_commit_id(ref, root),
+    error = function(e) {
+      stop(sprintf("Failed to locate git ref %s", ref))
```

Comment:
Is the default error from gert not good enough here?

Reply:
It absolutely is